src/app/shared/components/image-carousel/image-carousel.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit, Input, ViewChild, OnChanges } from '@angular/core';
+import { imgPath } from 'src/app/shared/models/carousel.model';
+
+@Component({
+  selector: 'app-image-carousel',
+  templateUrl: './image-carousel.component.html',
+  styleUrls: ['./image-carousel.component.scss']
+})
+export class ImageCarouselComponent implements OnInit, OnChanges {
+
+  @Input() images:imgPath[] = [];
+  @ViewChild('myCarousel', {static: false}) myCarousel;
+  constructor() { }
+
+  ngOnInit(): void {  
+  }
+
+  ngOnChanges() {
+    if (this.myCarousel) {
+      this.myCarousel.utils.carouselProperties.cellsElement.style.transform = 'none'
```

Comment:
;

Reply:
not clear why do we need change styles in ts

Reply:
> not clear why do we need change styles in ts

When user loads workshop-details page for the first time the tree of dom elements inside Carousel component get created. It looks like this:

Dots and arrows divs get created if we have more than one image in carousel.

![carousel bug](https://user-images.githubusercontent.com/37307672/150325851-7ab332ea-7239-448b-b2a5-0749d7906aa8.png)

Case 1: initially workshop with 1 image loads, carousel with one image get created, **Carousel.dotsArr = [1]**, div.carousel-cells style property has default values ----------->
user navigates to another workshop with 3 images: images load correctly, arrows get created, but **Carousel.dotsArr = [1]** and user sees only **one** dot instead of three; when he clicks arrow to slide to the next image,  **div.carousel-cells gets this style property**: 
	transform: translateX(-788px); //some pixels here 
	transition: transform 200ms ease-out 0s;
------------------>
then user goes to another workshop with one image: 
**Carousel.dotsArr = [1]**,  
**div.carousel-cells.style:**
        transform: translateX(-788px); //some pixels here 
	transition: transform 200ms ease-out 0s;

Because this style properties stays the same, user doesn’t see image in carousel because it lays (--788px) out of the screen
So carousel stays broken until user reload the page